torchvision/models/_api.py

```diff
@@ -1,19 +1,52 @@
+from __future__ import annotations
+
 import importlib
 import inspect
 import sys
 from dataclasses import dataclass, fields
 from inspect import signature
 from types import ModuleType
-from typing import Any, Callable, cast, Dict, List, Mapping, Optional, TypeVar, Union
+from typing import Any, Callable, cast, Dict, List, Mapping, Optional, Tuple, TypeVar, Union
 
 from torch import nn
 
 from torchvision._utils import StrEnum
 
 from .._internally_replaced_utils import load_state_dict_from_url
 
-
-__all__ = ["WeightsEnum", "Weights", "get_model", "get_model_builder", "get_model_weights", "get_weight", "list_models"]
+__all__ = [
+    "TransformsFactory",
+    "WeightsEnum",
+    "Weights",
+    "get_model",
+    "get_model_builder",
+    "get_model_weights",
+    "get_weight",
+    "list_models",
+]
+
+
+@dataclass(init=False)
+class TransformsFactory:
+    fn: Callable[..., nn.Module]
+    args: Tuple[Any, ...]
+    kwargs: Dict[str, Any]
+
+    def __init__(self, fn: Callable[..., nn.Module], *args: Any, **kwargs: Any) -> None:
+        self.fn = fn
+        self.args = args
+        self.kwargs = kwargs
```

Comment:
Wouldn't the dataclass automatically generate this `__init__()`?

Reply:
Nope, it would create

```python
def __init__(self, fn, args, kwargs):
    ...
```

Note the missing tuple and dict unpacking. I wanted this to make `TransformsFactory` as close to `functools.partial` as possible. Otherwise, we need to do something like `TransformsFactory(ImageClassification, (), dict(crop_size=[224]))` rather than `TransformsFactory(ImageClassification, crop_size=[224])`